lib/groonga/groonga-log/parser.rb

```diff
@@ -0,0 +1,86 @@
+# Copyright (C) 2017 Yasuhiro Horimoto <horimoto@clear-code.com>
+# Copyright (C) 2017 Kentaro Hayashi <hayashi@clear-code.com>
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+module Groonga
+  module GroongaLog
+    class Parser
+      PATTERN =
+        /\A(?<year>\d{4})-(?<month>\d\d)-(?<day>\d\d) (?<hour>\d\d):(?<minutes>\d\d):(?<seconds>\d\d)\.(?<micro_seconds>\d+)\|(?<log_level>.)\|(?<context_id>.+?)\|(?<message>.*)/
+
+      def parse(input)
+        return to_enum(:parse, input) unless block_given?
+
+        input.each_line do |line|
+          next unless line.valid_encoding?
+          m = PATTERN.match(line)
+
+          year = Integer(m['year'])
+          month = Integer(m['month'])
+          day = Integer(m['day'])
+          hour = Integer(m['hour'])
+          minutes = Integer(m['minutes'])
+          seconds = Integer(m['seconds'])
+          micro_seconds = Integer(m['micro_seconds'])
+          log_level = log_level_to_symbol(m['log_level'])
+          context_id = m['context_id']
+          message = m['message']
+          timestamp = Time.local(year, month, day,
```

Comment:
`Time.local` is always forced users to use their local time zone.
Should we provide UTC or user specific time zone parsing option?
If Groonga log contents is always written with local timezone, this comment seems to be useless....

Reply:
Groonga always uses local time.

Reply:
Hm, I see.... :thinking:

Fluentd plugins sometimes can choose UTC or local time for using time.
Current implementation looks good to me.